app/services/FeatureService.php

```diff
@@ -16,6 +16,47 @@ static function enabled(Feature $f, $uid = null)
         if (env("FEATURE_" . $f->value) == 'true')
             return true;
         $uid ??= User::getMainUserId();
-        return UserFeature::hasFeature($uid, $f->value);
+        return self::hasFeature($uid, $f->value);
+    }
+
+    /**
+     * Get all feature from a user : first check the ClubFeatures level then the UserFeatures level
+     * @return bool
+     */
+    static function hasFeature($uid, $feature)
+    {
+        $club_slug = ClubManagementService::getSelectedClubSlug();
+        if (
+            em()->createQuery("SELECT count(f) FROM ClubFeature f WHERE f.club = :c AND f.featureName = :feature AND f.enabled = :enabled")
+                ->setParameters(["c" => $club_slug, "feature" => $feature, "enabled" => true])
+                ->getSingleScalarResult()
+        ) {
+            $is_disabled = em()->createQuery("SELECT count(f) FROM UserFeature f WHERE f.user = :u AND f.featureName = :feature AND f.enabled = :enabled")
+                ->setParameters(["u" => $uid, "feature" => $feature, "enabled" => false])
+                ->getSingleScalarResult();
+            return (!$is_disabled);
+        }
+        return false;
+    }
+
+    /**
+     * list all feature for one user
+     * @return UserFeature[]
+     */
+    static function list_user($uid)
```

Comment:
Don't mix snake_case and camelCase method names in the same class

Reply:
Changed that, thx

Reply:
Nice thank you!